src/Payum/Core/Tests/CoreGatewayFactoryTest.php

```diff
@@ -72,6 +75,72 @@ public function shouldAlwaysAddHttpClientAsApi()
     /**
      * @test
      */
+    public function shouldCreateDefaultHttplugMessageFactory()
+    {
+        $factory = new CoreGatewayFactory();
+
+        $config = $factory->createConfig([]);
+        $this->assertArrayHasKey('httplug.message_factory', $config);
+        $this->assertInstanceOf(\Closure::class, $config['httplug.message_factory']);
+
+        $rc = new \ReflectionClass(CoreGatewayFactory::class);
+        $config = ArrayObject::ensureArrayObject($config);
+
+        $method = $rc->getMethod('buildClosures');
+
+        $rc->newInstance();
+        $method->setAccessible(true);
+        $method->invoke($factory, $config);
```

Comment:
the code above is too tricky. It is not acceptable for tests.

Could you do required stuff explicitly:
```php
<?php
$this->assertInstanceOf(\Closure::class, $config['httplug.message_factory']);
call_user_func($config['httplug.message_factory']);

```

hope you get the idea?